lib/circuit.js

```diff
@@ -187,6 +187,25 @@ class CircuitBreaker extends EventEmitter {
     }
   }
 
+  /**
+   * Shuts down this circuit breaker. All subsequent calls to the
+   * circuit will fail, returning a rejected promise.
+   */
+  shutdown () {
+    this.disable();
+    this.removeAllListeners();
+    this.hystrixStats.shutdown();
+    this[HYSTRIX_STATS] = undefined;
```

Comment:
Unsetting the hystrix stats to denote the breaker is shut down feels odd, especially when there is already a concept of state within the class. `this[STATE] = SHUTDOWN;` comes across as more meaningful/explicit.